plugin/evm/vm.go

```diff
@@ -170,6 +170,77 @@ func init() {
 	log.Root().SetHandler(log.LvlFilterHandler(log.LvlDebug, log.StreamHandler(originalStderr, log.TerminalFormat(false))))
 }
 
+type exportTxFunc func(assetID ids.ID, // AssetID of the tokens to export
+	amount uint64, // Amount of tokens to export
+	chainID ids.ID, // Chain to send the UTXOs to
+	to ids.ShortID, // Address of chain recipient
+	baseFee *big.Int, // fee to use post-AP3
+	keys []*crypto.PrivateKeySECP256K1R, // Pay the fee and provide the tokens
+) (*Tx, error)
+
+type parseAddressFunc func(addrStr string) (ids.ID, ids.ShortID, error)
+
+type AtomicTransactor struct {
+	sync.Mutex
+
+	avaxAssetID  ids.ID
+	parseAddress parseAddressFunc
+	newExportTx  exportTxFunc
+
+	pendingAtomicTxs []*Tx // atomic transactions that have yet to be committed
+}
+
+func newAtomicTransactor(avaxAssetID ids.ID, parseAddress parseAddressFunc, newExportTx exportTxFunc) *AtomicTransactor {
+	return &AtomicTransactor{
+		avaxAssetID:  avaxAssetID,
+		parseAddress: parseAddress,
+		newExportTx:  newExportTx,
+	}
+}
+
+func (a *AtomicTransactor) CreateExportTx(assetID string, gWeiAmount *big.Int, recipient string, baseFee *big.Int, keys []*crypto.PrivateKeySECP256K1R) error {
+	parsedAssetID, err := a.parseAssetID(assetID)
+	if err != nil {
+		return err
+	}
+
+	if gWeiAmount.Sign() != 1 {
+		return errors.New("argument 'amount' must be > 0")
+	}
+
+	chainID, to, err := a.parseAddress(recipient)
+	if err != nil {
+		return err
+	}
+
+	// I think I need to convert gWei to nAVAX here, then in UnsignedExportTx.EVMStateTransfer, it will be multiplied back to gWei.
```

Comment:
I believe converting from gWei to nAVAX at this point is correct, as we go the other direction in `EVMStateTransfer`. I'd appreciate validation on that.